src/js/components/TabForm.js

```diff
@@ -49,18 +54,30 @@ class TabForm extends React.Component {
     this.setState({currentTab});
   }
 
+  handleFormError() {
+    this.internalStorage_update({isFormValidated: false});
+  }
```

Comment:
Why can't we just `this.props.onError` here instead of setting this flag and moving the error-handling logic to `this.handleExternalSubmit` ? Imho `onError` should be the only handler invoked when an error occurred, and the parent component should be responsible to decide wether or not to proceed with submit. This feels unintuitive to me.